mixxx: build fails on aarch64-linux

### Nixpkgs version

- Stable (25.05)

### Describe the bug

Build fails with an [error due to warnings](https://github.com/user-attachments/files/21069746/error.txt)

### Steps to reproduce

`nix shell nixpkgs#mixxx` on an aarch64-linux machine

### Expected behaviour

Build works

### Screenshots

_No response_

### Relevant log output

```console
FAILED: CMakeFiles/DjInterop.dir/src/djinterop/util/chrono.cpp.o 
/nix/store/290idwd3a9c2lyi5mzixyn5qrxmzihzn-gcc-wrapper-14.3.0/bin/g++ -DDJINTEROP_SOURCE -DDjInterop_EXPORTS -I/build/source/build/include -I/build/source/include -I/build/source/SYSTEM -I/build/source/ext/sqlite_modern_cpp -I/build/source/ext/date -O3 -DNDEBUG -std=c++20 -fPIC -fvisibility=hidden -Wall -Wpedantic -Wno-unused-variable -Wno-unused-but-set-variable -Werror -MD -MT CMakeFiles/DjInterop.dir/src/djinterop/util/chrono.cpp.o -MF CMakeFiles/DjInterop.dir/src/djinterop/util/chrono.cpp.o.d -o CMakeFiles/DjInterop.dir/src/djinterop/util/chrono.cpp.o -c /build/source/src/djinterop/util/chrono.cpp
In file included from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/bits/exception_ptr.h:41,
                 from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/exception:166,
                 from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/ios:41,
                 from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/istream:40,
                 from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/sstream:40,
                 from /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/chrono:45,
                 from /build/source/src/djinterop/util/chrono.hpp:20,
                 from /build/source/src/djinterop/util/chrono.cpp:18:
In function 'constexpr std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = char]',
    inlined from 'constexpr void std::iter_swap(_ForwardIterator1, _ForwardIterator2) [with _ForwardIterator1 = char*; _ForwardIterator2 = char*]' at /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/bits/stl_algobase.h:185:11,
    inlined from 'constexpr void std::__reverse(_RandomAccessIterator, _RandomAccessIterator, random_access_iterator_tag) [with _RandomAccessIterator = char*]' at /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/bits/stl_algo.h:1062:18,
    inlined from 'constexpr void std::reverse(_BIter, _BIter) [with _BIter = char*]' at /nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/bits/stl_algo.h:1089:21,
    inlined from 'void date::detail::read(std::basic_istream<_CharT, _Traits>&, int, Args&& ...) [with CharT = char; Traits = std::char_traits<char>; Args = {char&, const char&}]' at /build/source/ext/date/date.h:6527:21,
    inlined from 'void date::detail::read(std::basic_istream<_CharT, _Traits>&, CharT, Args&& ...) [with CharT = char; Traits = std::char_traits<char>; Args = {int&, char&, const char&}]' at /build/source/ext/date/date.h:6488:9:
/nix/store/qzl7clzvis5z0b03gq9v67v1m4i8k68x-gcc-14.3.0/include/c++/14.3.0/bits/move.h:235:11: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  235 |       __a = _GLIBCXX_MOVE(__b);
      |           ^
```

### Additional context

_No response_

### System metadata

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 6.14.8-asahi, NixOS, 25.05 (Warbler), 25.05.20250629.b43c397`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/j95fcik6rzsydwips4m89dmlvfj9hg9y-source`
```

### Notify maintainers

@benley @bfortz 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/mixxx/default.nix
+++ b/pkgs/applications/audio/mixxx/default.nix
@@ -65,6 +65,12 @@ stdenv.mkDerivation rec {
     hash = "sha256-dKk3n3KDindnLbON52SW5h4cz96WVi0OPjwA27HqQCI=";
   };
 
+  # Should be removed when bumping to 2.6.x
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-warn "LIBDJINTEROP_VERSION 0.24.3" "LIBDJINTEROP_VERSION 0.26.1"
+  '';
+
   nativeBuildInputs = [
     cmake
     pkg-config

--- a/pkgs/by-name/li/libdjinterop/package.nix
+++ b/pkgs/by-name/li/libdjinterop/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   boost,
   cmake,
   ninja,
@@ -21,6 +22,14 @@ stdenv.mkDerivation (finalAttrs: {
     rev = finalAttrs.version;
     hash = "sha256-HwNhCemqVR1xNSbcht0AuwTfpRhVi70ZH5ksSTSRFoc=";
   };
+
+  patches = [
+    # https://github.com/xsco/libdjinterop/pull/161
+    (fetchpatch2 {
+      url = "https://github.com/xsco/libdjinterop/commit/94ce315cd5155bd031eeccfec12fbeb8e399dd14.patch";
+      hash = "sha256-WahMsFeetSlHHiIyaC04YxTiXDxD1ooASqoIP2TK9R0=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake

--- a/pkgs/by-name/li/libdjinterop/package.nix
+++ b/pkgs/by-name/li/libdjinterop/package.nix
@@ -10,16 +10,16 @@
   zlib,
 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   name = "libdjinterop";
 
-  version = "unstable";
+  version = "0.26.1";
 
   src = fetchFromGitHub {
     owner = "xsco";
     repo = "libdjinterop";
-    rev = "0.24.3";
-    hash = "sha256-Fp7AwEOq2JOpL60GDAXmA15ptFeLoG79nnnPXHl1Hjw=";
+    rev = finalAttrs.version;
+    hash = "sha256-HwNhCemqVR1xNSbcht0AuwTfpRhVi70ZH5ksSTSRFoc=";
   };
 
   nativeBuildInputs = [
@@ -46,4 +46,4 @@ stdenv.mkDerivation {
     maintainers = with maintainers; [ benley ];
     platforms = platforms.unix;
   };
-}
+})

